Taarafo.Core/Services/Foundations/Posts/PostService.Exceptions.cs

```diff
@@ -96,7 +104,7 @@ private IQueryable<Post> TryCatch(ReturningPostsFunction returningPostsFunction)
         }
 
         private PostDependencyException CreateAndLogCriticalDependencyException(
```

Comment:
Should this not be as follows?   It looks like the exception is downgraded to a PostDependencyException here and then you could just have called CreateAndLogDependencyException instead?  I think it should be PostCriticalDependencyException?

```
private PostCriticalDependencyException CreateAndLogCriticalDependencyException(
            Exception exception)
        {
            var postCriticalDependencyException = new PostCriticalDependencyException(exception);
            this.loggingBroker.LogCritical(postCriticalDependencyException);

            return postCriticalDependencyException;
        }
```

Reply:
the name that you mention that it's not explaining what our function does,

Reply:
@jhroodt152 there's no such type as `PostCriticalDependencyException` the `Critical` part is how we log it. It could be the same exception type but based on severity or circumstances we decide to make it critical or not.

Reply:
@cacti79 / @hassanhabib ,  thank you for explaining. I understand now. I was going through the unit tests and the method said ShouldThrow**CriticalDependency**Exception**OnModifyIfSqlErrorOccursAndLogItAsync, but when I followed the unit test, the sql exception gets localised as a **FailedPostStorageException** and when it comes to CreateAndLog**Critical**Dependency**Exception(failedPostStorageException) then method throws logs and throws a PostDependencyException.   Looking at this I though we might need to change this and created another exception called PostCriticalDependencyException. Following your explanation, I think the name of the unit test was what was misleading for me as I did not the Critical to the log level.   I'm catching up and learning a lot on this. Thank you.